### ISSUE TITLE
Private repo with TeamCity issue: Permission denied (publickey)

### ISSUE THREAD

**Issue by rbuussyghin** [danger/danger#266]
Here is my setup:

* Private repo on GitHub
* TeamCity uses SSH key to clone repo on build machine
* Danger uses access token to check pull request

During the build I see following error messages:

```
Starting: /bin/sh /Users/teamcity/buildAgent/temp/agentTmp/build4061647261074985101.sh
in directory: /Users/teamcity/buildAgent/work/5b721f8dcbe1cd7a
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Raising error 'Foo Error'

Danger has failed this build. 
Found 1 error and I don't have write access to the PR set a PR status.
Process exited with code 1
```

Despite _Found 1 error and I don't have write access to the PR set a PR status_ error message Danger was able to left a comment on pull request.

I've checked Danger source code and found [potential source of this problem](https://github.com/danger/danger/blob/master/lib/danger/request_source/github.rb#L49):

```ruby
      def setup_danger_branches
        # [ cut ]
        self.scm.exec "fetch origin +refs/pull/#{pull_id}/merge:#{EnvironmentManager.danger_head_branch}"
      end
```

Fetching origin like that means using `git` binary on a build machine without providing any auth information I specified in TeamCity (SSH key). Moreover, TeamCity always fetches most recent commit from the branch and I can't find any reason to fetch commits again.

To sum up:

* Incorrect error message _Found 1 error and I don't have write access to the PR set a PR status_. Danger actually has write access to the PR, she just don't have fetch access to the repo
* Fetching repo without any auth information

I'd like to discuss possible solutions to this issue :)

**Comment by justMaku** [comment]
Existence of this fetch was discussed and  well explained in #213. I'm all up for this fetch being optional though. What do you think @orta?

**Comment by orta** [comment]
Aye, I think we can change that. I'm happy for the commit range to be:

- base: Pulled in from GitHub or BitBucket etc, `pr_json["base"][ref"]`
- head: Current head

Meaning we can remove the `refs/pull/#{pull_id}/merge...` stuff.
